### ISSUE TITLE
MGLMapView calls MGLMapViewDelegate 'viewForAnnotation:' for annotations outside visible area

### ISSUE THREAD

**Issue by julaak** [mapbox/mapbox-gl-native#6173]
**Platform: iOS**
**Mapbox SDK version: 3.3.4**
### Steps to trigger behavior
1. Build & run Mapbox iOS example app located at 'platform/ios/ios.xcodeproj'
2. Click 'gear' icon on top left corner to open Map Settings menu
3. Click 'Add 100 Views' to add annotations to map
4. Pan map so annotations are outside the visible map area.
5. Keep panning into areas with no annotations
### Expected behavior

No annotations are visible on the map or nowhere near the borders of visible area, MGLMapView should not(?) call method:
**'- (MGLAnnotationView *)mapView:(MGLMapView *)mapView viewForAnnotation:(id<MGLAnnotation>)annotation'**

Why? Because there's no annotations visible.
### Actual behavior

**'- (MGLAnnotationView *)mapView:(MGLMapView *)mapView viewForAnnotation:(id<MGLAnnotation>)annotation'** is called constantly while panning on the map while there are no annotations nowhere near the visible area.

Below is the callstack.

![screen shot 2016-08-26 at 16 59 07](https://cloud.githubusercontent.com/assets/18900605/18007887/7fdb04d0-6bae-11e6-8f0e-7e2ee55c0fd5.png)

In a nutshell 'MGLMapView notifyMapChange:' method is called with param MapChangeDidFinishRenderingFrameFullyRendered which triggers a update for all annotations(?)
### Why is this a problem?

The map becomes very unresponsive with large amount of MGLAnnotationView objects placed on the map. I feel like the annotation view reuse queue is not working like it is intended to work.

**Comment by 1ec5** [comment]
Related: https://github.com/mapbox/mapbox-gl-native/issues/5787#issuecomment-235052750

/cc @boundsj @incanus

**Comment by boundsj** [comment]
This is fixed in https://github.com/mapbox/mapbox-gl-native/pull/5987. #5987 will be in the next release (beta 2).
